### PR TITLE
[RLlib; Offline RL] - Replace GAE in `MARWILOfflinePreLearner` with `GeneralAdvantageEstimation` connector in learner pipeline.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

In `PPO` the GAE in the `Learner` had been replaced by the `GeneralAdvantageEstimation` connector in its learner pipeline. `MARWIL` basically needs the same calculation albeit not directly in the learner but in its `PreLearner`. This PR proposes this change.

- It adds the `AddOneTsToEpisodesAndTruncate` and `GeneralAdvantageEstimation` to `MARWIL`'s learner pipeline.
- Deprecates the `MARWILOfflinePreLearner` b/c with the modified learner pipeline 'MARWIL' can fall back onto the default `OfflinePreLearner`.
- Changed the default for episodes in offline batches from `finalize=False` to `finalize=True` b/c this is needed for the `AddOneTsToEpisodesAndTruncate` connector.

## Related issue number

Related to #47401

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/marwil/marwil.py [hunk 1]
```diff
@@ -283,13 +277,27 @@ def build_learner_connector(
             device=device,
         )
 
+        # Before anything, add one ts to each episode (and record this in the loss
```
Comment (sven1977) [inline rllib/algorithms/marwil/marwil.py:280]: I like the Learner connector Pipeline more and more every day. It's super super powerful. We just have to do a good job now documenting this (on me) and making sure users know how to debug into it (by simply switching off tune and setting `num_learners=0`).
🙌

File: rllib/algorithms/marwil/marwil_offline_prelearner.py [hunk 2]
```diff
@@ -1,237 +0,0 @@
-import numpy as np
```
Comment (sven1977) [inline rllib/algorithms/marwil/marwil_offline_prelearner.py:1]: no way!! awesome to be able to delete this.

File: rllib/tuned_examples/marwil/cartpole_marwil.py [hunk 3]
```diff
@@ -48,7 +48,7 @@
     # as remote learners.
     .offline_data(
         input_=[data_path.as_posix()],
-        input_read_method_kwargs={"override_num_blocks": max(args.num_gpus, 1)},
+        input_read_method_kwargs={"override_num_blocks": max(args.num_gpus, 2)},
```
Comment (sven1977) [inline rllib/tuned_examples/marwil/cartpole_marwil.py:51]: nit nit: Can we explain in one line, why 2 should be the max here?

Comment (sven1977) [reply rllib/tuned_examples/marwil/cartpole_marwil.py:51]: probably same explanation as in pendulum CQL above?

File: rllib/offline/offline_prelearner.py [hunk 4]
```diff
@@ -412,7 +412,7 @@ def _map_sample_batch_to_episode(
                 )
                 # Create a `SingleAgentEpisode`.
                 episode = SingleAgentEpisode(
-                    id_=batch[schema[Columns.EPS_ID]][i][0],
+                    id_=str(batch[schema[Columns.EPS_ID]][i][0]),
```
Comment (sven1977) [inline rllib/offline/offline_prelearner.py:415]: nice catch

**Review by sven1977 — APPROVED** [review]
Approved! Nice consolidation/cleanup PR. Good stuff @simonsays1980 !

**Comment by sven1977** [comment]
One nit nit (see comments).

File: rllib/offline/offline_prelearner.py (hunk 4 above)
Comment (simonsays1980) [reply rllib/offline/offline_prelearner.py:415]: Yeah, well, I got an error running the connector pipeline.

**Pull request merged**